tarpc/tests/dataservice.rs

```diff
@@ -0,0 +1,52 @@
+use futures::{future, prelude::*};
+use std::io;
+use tarpc::{
+    client, context,
+    server::{self, Handler},
+};
+
+#[tarpc::derive_serde]
+#[derive(PartialEq)]
+pub enum TestData {
+    Black,
+    White,
+}
+
+#[tarpc::service]
+pub trait ColorProtocol {
+    async fn get_opposite_color(color: TestData) -> TestData;
+}
+
+#[derive(Clone)]
+struct ColorServer;
+
+#[tarpc::server]
+impl ColorProtocol for ColorServer {
+    async fn get_opposite_color(self, _: context::Context, color: TestData) -> TestData {
+        match color {
+            TestData::White => TestData::Black,
+            TestData::Black => TestData::White,
+        }
+    }
+}
+
+#[tokio::test]
+async fn test_call() -> io::Result<()> {
+    let (client_transport, server_transport) = tarpc::transport::channel::unbounded();
```

Comment:
This channel is an in-memory channel and does't perform serialization, so I don't think it's actually testing that the serde derive works correctly.

Reply:
Oh, i'll change that to a serde based channel then :)